dist/$rsi.schema.json

```diff
@@ -0,0 +1,227 @@
+          }
+        },
+        "types": {
+          "$ref": "#/definitions/typesDefinition"
+        },
+        "description": {
+          "type": "string"
+        },
+        "$schema": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "description",
+        "name",
+        "$schema"
+      ],
+      "additionalProperties": false
+    },
+    "resourcesDefinition": {
+      "type": "object",
+      "properties": {
+        "description": {
+          "type": "string"
+        },
+        "objectModel": {
+          "$ref": "#/definitions/objectModelDefinition"
+        }
+      },
+      "required": [
+        "description"
+      ],
+      "additionalProperties": true
+    },
+    "typesDefinition": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "description": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "description"
+      ],
+      "additionalProperties": true
+    },
+    "objectModelDefinition": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "object",
+          "properties": {
+            "description": {
+              "type": "string",
+              "minLength": 2
+            },
+            "type": {
+              "type": "string",
+              "enum": [
+                "string",
+                "number",
+                "integer",
+                "boolean",
+                "array",
+                "object"
+              ]
+            },
+            "format": {
+              "type": "string",
+              "pattern": "uuid"
+            }
+          },
+          "required": [
+            "description",
+            "type",
+            "format"
+          ]
+        },
+        "name": {
+          "type": "string"
+        },
+        "uri": {
+          "type": "object",
+          "properties": {
+            "description": {
+              "type": "string",
+              "minLength": 2
+            },
+            "type": {
+              "type": "string",
+              "pattern": "string"
+            },
+            "format": {
+              "type": "string",
+              "pattern": "uri"
+            }
+          },
+          "required": [
+            "description",
+            "type",
+            "format"
+          ]
+        }
+      },
+      "patternProperties": {
+        "^[a-zA-Z0-9]+$": {
+          "type": "object"
+        }
+      },
+      "additionalProperties": false,
+      "required": [
+        "id",
+        "name",
+        "uri"
+      ]
+    },
+    "objectModelReference": {
+      "type": "object",
+      "properties": {
+        "#ref": {
+          "type": "string",
+          "pattern": "^([a-zA-Z]+\\.[a-zA-Z]+\\.[a-zA-Z]+Object|[a-zA-Z]+\\.[a-zA-Z]+Type)$"
+        }
+      },
+      "additionalProperties": false,
+      "required": [
+        "#ref"
```

Comment:
How do you plan to validate a schema when for example a resource has two references, and both are required. I think, `#ref` is not enough in the `required` property. It should be the full path.